### ISSUE TITLE
[Platform Initiative] Wrap up and roll out dynamic environment building and sharing

### ISSUE THREAD

**Issue by Gman0909** [2i2c-org/infrastructure#5064]
## Context 

As hubs get a wider user base, different users need different environments to do their work. The same user may need multiple environments at different times too. Long term, we can not rely on admins to continuously update the small set of images available, and for users to have to keep up with modifying their code to work with newer packages. We need a way for users to be able to specify environments (without having to really understand Docker), with a view to later being able to share them with others in a standardized way. This should be done in a way that’s compatible with community standards as well, rather than re-inventing something new. As part of the previous PI, we have deployed [jupyterhub-fancy-profiles](https://github.com/yuvipanda/jupyterhub-fancy-profiles) to VEDA for an improved logged-in environment choosing experience. This builds on top of that, with the goal of making it easier for users to easily & reproducibly create and use whatever environments they want, and eventually share it with others as well (See #5087 ).

We propose to improve the [Dynamic Image Building](https://2i2c.org/blog/2024/jupyterhub-binderhub-gesis/) feature of jupyterhub-fancy-profiles (courtesy a prior grant from GESIS) until it can be put in front of production users. Once deployed, end users can easily create environments in the form of GitHub repositories (or Zenodo DOIs, or many other versioned sources) and, at some point in future, share it with each other (See #5087 ). These same environments will work on mybinder.org, our existing image building infrastructure and the ephemeral hub set up as part of JH-5 by using the same backend technology to build images ([binderhub](https://github.com/jupyterhub/binderhub/) and repo2docker).

This work has partially been done as part of the DevSeed collaboration, but needs to be completed. 

See related issues [here ](https://github.com/NASA-IMPACT/veda-jupyterhub/issues/58 ) and [here](https://github.com/NASA-IMPACT/veda-jupyterhub/issues/45)

The second part of this effort is captured in #5087 

## Why is this important

Dynamic Image Building is a powerful enabler for many community workflows, and getting it rolled out fully is key to ensuring our communities have friction-less access to it. 

## Definition of Done

- [ ] We have a clear UX allowing users to define, and build their custom environments
- [ ] The UX is tested and validated with end-users
- [x] It has been deployed into a production hub
- [x] It has been shown to be replicably deployable
- [ ] It is well documented
- [ ] It has been added to our product menu.
- [ ] Lessons learned have been documented
- [ ] We have communicated the availability of this feature to our communities
- [ ] We have communicated the availability of this features to our Sales function

**Comment by jnywong** [comment]
Candidate for deployment to production hub: https://github.com/2i2c-org/infrastructure/issues/5232

**Comment by yuvipanda** [comment]
This has also already been deployed to a few hubs - primarily, NASA VEDA and NASA MAAP

**Comment by Gman0909** [comment]
@yuvipanda fantastic. Can you please go through the DoD so we can see what’s outstanding before we can call this done?

**Comment by yuvipanda** [comment]
There hasn't been any UX validation done at all yet. There also isn't any docs or best practices on how to use it.

**Comment by yuvipanda** [comment]
https://github.com/NASA-IMPACT/veda-docs/issues/186 is VEDA's attempt at writing docs

**Comment by Gman0909** [comment]
@yuvipanda I can add it, but I see a lot of the DoD is still incomplete, and I would wait until that isn't the case before adding the feature and calling it done. Also, is there a schedule for deploying to the hubs listed in the sub-tasks?

**Comment by Gman0909** [comment]
Given the work already completed by DevSeed, including work for #5087, @yuvipanda will re-scope this initiative into tightly scoped set of outstanding work needed to consider these features "rolled out"

**Comment by yuvipanda** [comment]
@Gman0909 tracked that work in https://github.com/2i2c-org/infrastructure/issues/6333

**Comment by jmunroe** [comment]
@yuvipanda Could
- https://github.com/2i2c-org/infrastructure/issues/6512

be considered up as part of this initiative?

**Comment by yuvipanda** [comment]
This should be replaced with a number of smaller and more focused initiatives. https://github.com/2i2c-org/initiatives/issues/2 is one.

**Comment by Gman0909** [comment]
@yuvipanda closing this in favour of your revised initiatives - please take a look at the sub-tasks and re-parent, close or move as appropriate
